pkg/sources/errors.go

```diff
@@ -20,6 +20,10 @@ func NewScanErrors() *ScanErrors {
 
 // Add an error to the collection in a thread-safe manner.
 func (s *ScanErrors) Add(err error) {
+	if err == nil {
+		return
+	}
```

Comment:
This might not be correct. Do we want `ScanErrors` to hold nil?

Reply:
I can't imagine why we would want to hold nils but perhaps I'm wrong

Reply:
Calling @mcastorina to double check the intention here, since the test explicitly contained that case.

Reply:
I agree, it should probably never have nil errors. I probably added that case to document existing behavior.